src/test/java/test/TestAPI.java

```diff
@@ -210,15 +176,27 @@ public void validateFindingExistingUserBySurname() {
     @Test
     public void validateFindingUserBySurnameThatDoesntExist() {
         String invalidSurname = "xyz";
+
         RequestSpecification request = given()
                 .param("surname", invalidSurname);
 
-        Response response = given()
-                .spec(request)
-                .when()
-                .get(baseURI + USER_SURNAME_ENDPOINT);
+        Response response = getUserListUsingSurname(request);
 
         Assertions.assertFalse(response.body().asString().contains(invalidSurname));
         Assertions.assertEquals(200, response.statusCode());
     }
+
+    @Test
+    public void validateAddingAddressAsUser() throws JsonProcessingException {
+        Address address = Address.builder().withTestValues().randomId().build();
+
+        RequestSpecification request = given()
+                .header("Content-Type", "application/json")
+                .body(address.toJsonString());
+
+        Response response = postUser(request);
+
+        Assertions.assertEquals(500, response.statusCode());
+        Assertions.assertFalse(db.existsAddress(address.getId()));
+    }
```

Comment:
I don't think this test is meaningful from software testing perspective, let's remove it.